### DISCUSSION TITLE
File creation template

### DISCUSSION THREAD

**Discussion by lcp5y3** [helix-editor/helix#12233] [Ideas]
# File creation template

As former user of Vim, one thing that I miss, is templates/lang.tpl.
This template are used to creating a new file with default values.

Example "*py.tpl*:

```python
#!/usr/bin/env python3
#encoding utf8 

"""
project name:
date: FDATE
author:
description:


if __name__ == "__main__":
    print("Hello")
```

The idea, if I type the command *:open main.py* and there is no *main.py* available in path, it will use the py.tpl to create a new one.

**Comment by el-pendeloco** [comment]
Yeah I'd like that too. Something more granular though, because I don't think one template per language would cut it. For example, I'd like to have a template for a new typescript web component, one for a wrapped Zod schema, and so on and so forth.
Ideally, it would support passing variables and therefore running a templating engine under the hood.
I'd love that feature and it would probably require a bit more work (with many templates possible) then mentioned.

Maybe a new command: `:from-template element.tpl toto.ts -v component-name=BigComponent`, although not sure about the syntax. This will be possible with the upcoming extension plugins, I believe. So maybe not something for core helix :shrug:

**Reply by kumawatdarshan** [reply]
sounds like plugin territory
